index.js

```diff
@@ -1122,8 +1136,8 @@
               }
             } else {
               okTypes = determineActualTypesStrict(values);
-              if (isEmpty(okTypes)) {
-                throw conflictingTypeVar(
+              if (isEmpty(okTypes) && !isEmpty(values)) {
```

Comment:
This check and the `!isEmpty(okTypes)` check below were removed—perhaps erroneously—in #38.